snorkel/analysis/utils.py

```diff
@@ -0,0 +1,251 @@
+    """
+    return np.argmax(probs, axis=1) + 1
+
+
+def preds_to_probs(preds: np.ndarray, num_classes: int) -> np.ndarray:
+    """Convert an array of probabilistic labels into an array of predictions
+
+    Parameters
+    ----------
+    pred
+        A [num_datapoints] or [num_datapoints, 1] array of predictions
+
+    Returns
+    -------
+    np.ndarray
+        A [num_datapoints, num_classes] array of probabilistic labels with probability
+        of 1.0 in the column corresponding to the prediction
+    """
+    return np.eye(num_classes)[preds.squeeze() - 1]
+
+
+def arraylike_to_numpy(
+    array_like: ArrayLike, flatten: bool = True, cast_to_int: bool = True
+) -> np.ndarray:
+    """Convert an ArrayLike (e.g., list, tensor, etc.) to a numpy array
+
+    Also optionally flatten [n, 1] arrays to [n] and cast all values to ints.
+    This method is typically used to sanitize labels before use with analysis tools or
+    metrics that expect 1D numpy arrays as inputs.
+
+    Parameters
+    ----------
+    array_like
+        An Arraylike to convert
+    flatten
+        If True, flatten numpy array into a 1D array
+    cast_to_int
+        If True, cast all values to ints
+
+    Returns
+    -------
+    np.ndarray
+        The input converted to an np.ndarray
+
+    Raises
+    ------
+    ValueError
+        Provided input could not be converted to an np.ndarray
+    """
+
+    orig_type = type(array_like)
+
+    # Convert to np.ndarray
+    if isinstance(array_like, np.ndarray):
+        pass
+    elif isinstance(array_like, list):
+        array_like = np.array(array_like)
+    elif isinstance(array_like, sparse.spmatrix):
+        array_like = array_like.toarray()
+    elif isinstance(array_like, torch.Tensor):
+        array_like = array_like.numpy()
+    elif not isinstance(array_like, np.ndarray):
+        array_like = np.array(array_like)
+    else:
+        msg = f"Input of type {orig_type} could not be converted to an np.ndarray"
+        raise ValueError(msg)
+
+    # Correct shape
+    if flatten:
+        if (array_like.ndim > 1) and (1 in array_like.shape):
+            array_like = array_like.flatten()
+        if array_like.ndim != 1:
+            raise ValueError("Input could not be converted to 1d np.array")
+
+    # Convert to ints
+    if cast_to_int:
+        if any(array_like % 1):
+            raise ValueError("Input contains at least one non-integer value.")
+        array_like = array_like.astype(np.dtype(int))
+
+    return array_like
+
+
+def convert_labels(Y: ArrayLike, source: str, target: str) -> ArrayLike:
+    """Convert a matrix from one label type to another
+
+    Args:
+        Y: A np.ndarray or torch.Tensor of labels (ints) using source convention
+        source: The convention the labels are currently expressed in
+        target: The convention to convert the labels to
+    Returns:
+        Y: an np.ndarray or torch.Tensor of labels (ints) using the target convention
+
+    Conventions:
```

Comment:
Note for later: we should probably remove the `plus minus` notation completely (including tutorials, that's on me)

Reply:
I think we definitely shouldn't use it anywhere, but I think there's a good chance that datasets people bring in will have one-zero or plus-minus notation, and it's nice to have an easy way for them to convert into the form we accept.